index.ts

```diff
@@ -355,85 +363,76 @@ function doTransform(checker, targetVersion, k) {
           typeArguments[0],
           ts.factory.createTypeReferenceNode(ts.factory.createIdentifier("Exclude"), [
             ts.factory.createTypeOperatorNode(ts.SyntaxKind.KeyOfKeyword, typeArguments[0]),
-            typeArguments[1]
-          ])
+            typeArguments[1],
+          ]),
         ]);
       }
     } else if (semver.lt(targetVersion, "4.0.0") && n.kind === ts.SyntaxKind.NamedTupleMember) {
-      const member = /** @type {import("typescript").NamedTupleMember} */ (n);
+      const member = n as NamedTupleMember;
       return ts.addSyntheticLeadingComment(
         member.dotDotDotToken ? ts.factory.createRestTypeNode(member.type) : member.type,
         ts.SyntaxKind.MultiLineCommentTrivia,
         ts.unescapeLeadingUnderscores(member.name.escapedText),
-        /*hasTrailingNewline*/ false
+        /*hasTrailingNewline*/ false,
       );
     } else if (semver.lt(targetVersion, "4.7.0") && ts.isTypeParameterDeclaration(n)) {
       return ts.factory.createTypeParameterDeclaration(
         n.modifiers?.filter(
-          modifier => modifier.kind !== ts.SyntaxKind.InKeyword && modifier.kind !== ts.SyntaxKind.OutKeyword
+          (modifier) => modifier.kind !== ts.SyntaxKind.InKeyword && modifier.kind !== ts.SyntaxKind.OutKeyword,
         ),
         n.name,
         n.constraint,
-        n.default
+        n.default,
       );
     }
-    return ts.visitEachChild(n, transform, k);
+    return ts.visitEachChild(n, nodeVisitor, k);
+  };
+
+  return (sourceFile: SourceFile): SourceFile => {
```

Comment:
calling out that right here is the only functional change from the original that is more than just adding type safety -- previously this was a function bind, which worked because the node visitor when given the source file returned a source file node, but the type signature of the transform didn't guarantee this.